cmd/sandbox/start.go

```diff
@@ -188,31 +190,46 @@ func startSandbox(ctx context.Context, cli docker.Docker, reader io.Reader) (*bu
 }
 
 // Returns the alternate image if specified, else
-// if no version is specified then the Latest release of cr.flyte.org/flyteorg/flyte-sandbox:dind is used
+// if no version is specified then the Latest release of cr.flyte.org/flyteorg/flyte-sandbox:dind-{SHA} is used
 // else cr.flyte.org/flyteorg/flyte-sandbox:dind-{SHA}, where sha is derived from the version.
 func getSandboxImage(version string, alternateImage string) (string, error) {
-
 	if len(alternateImage) > 0 {
 		return alternateImage, nil
 	}
-
-	var tag = dind
-	if len(version) > 0 {
-		isGreater, err := util.IsVersionGreaterThan(version, sandboxSupportedVersion)
+	if len(version) == 0 {
+		var err error
+		releases, err := githubutil.GetListRelease(flyteRepository)
 		if err != nil {
 			return "", err
 		}
-		if !isGreater {
-			return "", fmt.Errorf("version flag only supported with flyte %s+ release", sandboxSupportedVersion)
-		}
-		sha, err := githubutil.GetSHAFromVersion(version, flyteRepository)
-		if err != nil {
-			return "", err
+		for _, v := range releases {
+			if *v.Prerelease && sandboxConfig.DefaultConfig.Prerelease {
+				logger.Infof(context.Background(), "sandbox started with pre release %s", *v.TagName)
+				version = *v.TagName
+				break
+			} else if !*v.Prerelease && !sandboxConfig.DefaultConfig.Prerelease {
+				logger.Infof(context.Background(), "sandbox started with release %s", *v.TagName)
+				version = *v.TagName
+				break
+			}
 		}
-		tag = fmt.Sprintf("%s-%s", dind, sha)
 	}
-
-	return docker.GetSandboxImage(tag), nil
+	selectedVersion, err := githubutil.CheckVersionExist(version, flyteRepository)
+	if err != nil {
+		return "", err
+	}
+	isGreater, err := util.IsVersionGreaterThan(*selectedVersion.TagName, sandboxSupportedVersion)
+	if err != nil {
+		return "", err
+	}
+	if !isGreater {
+		return "", fmt.Errorf("version flag only supported with flyte %s+ release", sandboxSupportedVersion)
+	}
+	sha, err := githubutil.GetSHAFromVersion(*selectedVersion.TagName, flyteRepository)
+	if err != nil {
+		return "", err
+	}
+	return docker.GetSandboxImage(fmt.Sprintf("dind-%s", sha)), nil
```

Comment:
L217-228 can all be abstracted out inside this function

Reply:
GetSandbox image can work directly off the version and you can even abstract the sha in there